server/chat/backend/agent/utils/immediate_save_handler.py

```
@@ -42,6 +42,45 @@ def handle_immediate_save(session_id: str, user_id: str, question: str) -> bool:
+
+        result = cursor.fetchone()
+        if result and result[0]:
+            messages = result[0] if isinstance(result[0], list) else []
+            # Remove the last message only if it's a user message
+            if messages and messages[-1].get('sender') == 'user':
+                messages.pop()
+                cursor.execute("""
+                    UPDATE chat_sessions
+                    SET messages = %s, updated_at = %s
+                    WHERE id = %s AND user_id = %s
+                """, (json.dumps(messages), datetime.now(), session_id, user_id))
```

Ruff (0.15.17)

[warning] `datetime.datetime.now()` called without a `tz` argument

(DTZ005)